app/javascript/components/workflows/professor/exams/new/editor/components/questions/CodeTag.tsx

```diff
@@ -179,85 +224,66 @@ const CodeTag: React.FC<CodeTagProps> = (props) => {
       references = partReferences.references;
       break;
     default:
-      throw new ExhaustiveSwitchError(choices);
+      throw new ExhaustiveSwitchError(choice);
   }
   const { fmap } = useContext(ExamContext);
   const filteredFiles = getFilesForRefs(fmap, references);
+  return (
+    <>
+      <Form.Label column sm={2}>Correct answer</Form.Label>
+      <Col sm={6}>
+        <CodeTagVal value={value} hideFile={countFiles(filteredFiles) === 1} />
+      </Col>
+      <Col sm={4}>
+        <Button
+          onClick={(): void => setShowModal(true)}
+        >
+          Choose line
+        </Button>
+        <FileModal
+          disabled={false}
+          references={references}
+          show={showModal}
+          onClose={(): void => setShowModal(false)}
+          onSave={(newState): void => {
+            setShowModal(false);
+            onChange(newState);
+          }}
+          startValue={value}
+        />
+      </Col>
+    </>
+  );
+};
+
+interface CodeTagProps {
+  qnum: number;
+  pnum: number;
+  bnum: number;
+}
+
+const CodeTag: React.FC<CodeTagProps> = (props) => {
+  const {
+    qnum,
+    pnum,
+    bnum,
+  } = props;
   return (
     <>
       <Prompted
         qnum={qnum}
         pnum={pnum}
         bnum={bnum}
-        prompt={prompt.value}
-        onChange={(newPrompt): void => {
-          if (onChange) {
-            onChange({ ...info, prompt: { type: 'HTML', value: newPrompt } }, value);
-          }
-        }}
       />
       <Form.Group as={Row} controlId={`${qnum}-${pnum}-${bnum}-source`}>
-        <Form.Label column sm={2}>Files source</Form.Label>
-        <Col sm={10}>
-          <ButtonGroup>
-            <Button
-              variant={choices === 'exam' ? 'secondary' : 'outline-secondary'}
-              active={choices === 'exam'}
-              onClick={(): void => {
-                onChange({ ...info, choices: 'exam' }, undefined);
-              }}
-            >
-              Files for full exam
-            </Button>
-            <Button
-              variant={choices === 'question' ? 'secondary' : 'outline-secondary'}
-              active={choices === 'question'}
-              onClick={(): void => {
-                onChange({ ...info, choices: 'question' }, undefined);
-              }}
-            >
-              Files for current question
-            </Button>
-            <Button
-              variant={choices === 'part' ? 'secondary' : 'outline-secondary'}
-              active={choices === 'part'}
-              onClick={(): void => {
-                onChange({ ...info, choices: 'part' }, undefined);
-              }}
-            >
-              Files for current part
-            </Button>
-          </ButtonGroup>
-        </Col>
+        <Fields names={['answer', 'choices']} component={EditChoices} />
```

Comment:
How does this work?  Obviously something redux-forms here that, what, reflectively takes the names given and produces a props object with their values somehow?

Reply:
`Field` grabs a key in the state relative to the last `FormSection`.
`Fields` grabs more than one at a time.
https://redux-form.com/8.3.0/docs/api/fields.md/